test/integration/apiserver/crd_validation_expressions_test.go

```diff
@@ -888,6 +973,76 @@ var structuralSchemaWithBlockingErr = []byte(`
   }
 }`)
 
+var structuralSchemaWithItemsUnderArray = []byte(`
+{
+  "openAPIV3Schema": {
+    "description": "CRD with CEL validators",
+    "type": "object",
+    "properties": {
+      "spec": {
+        "type": "object",
+        "properties": {
+          "backend": {
+            "type": "array",
+            "maxItems": 100,
+            "items": {
+              "type": "object",
+              "properties": {
+                "replicas": {
+                  "type": "integer"
+                }
+              },
+              "required": [
+                "replicas"
+              ],
+              "x-kubernetes-validations": [
+                {
+                  "rule": "0 <= self.replicas && self.replicas <= 10"
+                }
+              ]
+            }
+          }
+        }
+      }
+    }
+  }
+}`)
+
+var structuralSchemaWithItemsUnderObject = []byte(`
+{
+  "openAPIV3Schema": {
+    "description": "CRD with CEL validators",
+    "type": "object",
+    "properties": {
+      "spec": {
+        "type": "object",
+        "properties": {
+          "backend": {
+            "type": "object",
+            "maxItems": 100,
+            "items": {
```

Comment:
Is this just a short-term test case to exercise the problematic scenario for backporting? in the future, I'd like to explore how we can prevent this type of schema from being accepted at all (in a ratcheting way). Is there an issue tracking that?

Reply:
Future work would be tracked in https://github.com/kubernetes/kubernetes/issues/126273 :)